### ISSUE TITLE
Unable to get tags working

### ISSUE THREAD

**Issue by droddy** [plmwong/log4net-raygun#14]
I've put the following in Global.asax Application_Start() in my MVC project and still am not able to get Raygun messages with the tags--not sure what I'm missing:

`            log4net.GlobalContext.Properties[RaygunAppender.PropertyKeys.Tags] = "droddy|testing";
            log4net.LogicalThreadContext.Properties[RaygunAppender.PropertyKeys.Tags] = "droddy|testing";
`
These lines definitely get hit when I start debugging but the exception that shows up in Raygun has no Tags.

**Comment by plmwong** [comment]
Hi there,

I'm having trouble replicating. Could you please clarify for me, your exceptions/errors are showing up in Raygun, just without the tags attached to them?

E.g., you aren't getting anything similar to the following showing up:
![image](https://cloud.githubusercontent.com/assets/4966809/13268480/9148231e-da37-11e5-9b95-c609c113bf94.png)

If you are able to, could you please turn log4net internal debug/trace on (http://stackoverflow.com/questions/756125/how-to-track-down-log4net-problems) and see if log4net.Raygun is failing somewhere?

Thanks,
Phil

**Comment by droddy** [comment]
Turned on internal debug/trace for log4net and didn't see any failures in that log at all. Here's the log file in case you're interested:
[log4net.debuglog.txt](https://github.com/plmwong/log4net-raygun/files/162158/log4net.debuglog.txt)

**Comment by droddy** [comment]
this might have something to do with it. We're using SLF (Simple Logging Facade) to decouple our code from log4net-specific logging:
http://slf.codeplex.com/

**Comment by droddy** [comment]
@plmwong any ideas?

**Comment by plmwong** [comment]
Sorry about the delay and thanks for the extra info.

I'm not seeing anything raygun related in that debuglog file you have attached, I would expect there to be some lines showing that the RaygunAppender has been created by log4net, e.g.

```
log4net: Loading Appender [RaygunAppender] type: [log4net.Raygun.RaygunAppender, log4net.Raygun]
log4net: Setting Property [ApiKey] to String value [<redacted>]
log4net: Setting Property [Threshold] to Level value [ERROR]
log4net: Created Appender [RaygunAppender]
log4net: Adding appender named [RaygunAppender] to logger [root].
log4net: Hierarchy Threshold []
log4net: RaygunAppender: Received Logging Event with Logging Level 'ERROR'
```

Also, I quickly put together an ASP.NET MVC default project to test things out: [log4net-raygun-mvc.zip](https://github.com/plmwong/log4net-raygun/files/180210/log4net-raygun-mvc.zip)

Basically:
* Added the SLF and SLF Facade assemblies.
* Added log4net.Raygun.1.2.10 NuGet package, as SLF seems to be built against log4net 1.2.10 and I wanted to avoid assembly clashing.
* Global.asax.cs - Added tags using the lines supplied above.
* Web.config - Configured log4net.Raygun and SLF.
* Default.aspx.cs - Get SLF LoggingService and log an exception.

When I run the solution I am getting the tags showing:
![image](https://cloud.githubusercontent.com/assets/4966809/13888675/db15df24-ecff-11e5-8810-0d8dd918897a.png)

Hopefully this test project can help you out.

What version of log4net.Raygun are you using (and which package?).


-- Phil

**Comment by droddy** [comment]
@plmwong Wow. Thanks for all of your effort! 
I think you are onto something about the different log4net versions. We're using 
log4net.Raygun 4.2.0.18164 from Nuget. 
Log4Net 1.2.13.0
SLF 2.0.1.0
SLF.Log4NetFacade 2.0.1.0

**Comment by plmwong** [comment]
@droddy Were you able to figure out your issue?
